cmd/opm/alpha/template/composite.go

```diff
@@ -26,6 +32,37 @@ and a 'composite template' file`,
 and a 'composite template' file`,
 		Args: cobra.MaximumNArgs(0),
 		Run: func(cmd *cobra.Command, args []string) {
+			containerTool = "docker"
+			var tempCatalog io.ReadCloser
+			catalogURI, err := url.ParseRequestURI(catalogFile)
+			if err != nil {
+				tempCatalog, err = os.Open(catalogFile)
+				if err != nil {
+					log.Fatalf("opening catalog config file %q: %v", catalogFile, err)
+				}
+				defer tempCatalog.Close()
+			} else {
+				tempResp, err := http.Get(catalogURI.String())
+				if err != nil {
+					log.Fatalf("fetching remote catalog config file %q: %v", catalogFile, err)
+				}
+				tempCatalog = tempResp.Body
+				defer tempCatalog.Close()
+			}
+			catalogData := tempCatalog
```

Comment:
This all moves into the API: https://github.com/operator-framework/operator-registry/blob/7629c6f991fc8dcce91bcb5ee4e998678bbd7779/alpha/template/composite/composite.go#L88